File: WebApp/WebApp/Models/PriceHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace WebApp.Models
{
    public class PriceHistory
    {
        [Key]
        public Guid Id { get; set; }
        public Guid PricelistId { get; set; }
        public Pricelist Pricelist { get; set; }
        public Guid ProductTypeId { get; set; }
        public ProductType ProductType { get; set; }
        public Double Price { get; set; }
    }
}

File: angular-app/src/models/price-history.ts
import { Guid } from "guid-typescript";
import { Pricelist } from './pricelist';
import { ProductType } from './product-type';

export class PriceHistory {
    public id: Guid;
    public pricelistId: Guid;
    public pricelist: Pricelist;
    public productTypeId: Guid;
    public productType: ProductType;
    public price: number;
}